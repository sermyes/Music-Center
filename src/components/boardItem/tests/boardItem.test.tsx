import BoardItem from '../boardItem';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { fakePosts } from './../../tests/posts';

describe('Board Item', () => {
  const admin = fakePosts.admin!;
  const notice = fakePosts.notice!['1'];
  const question = fakePosts.question!['2'];
  const post = fakePosts.post!['3'];
  post.date = new Date().getTime();
  const onRemove = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <BoardItem post={post} admin={admin} onRemove={onRemove} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders notice type data', async () => {
    render(<BoardItem notice={notice} admin={admin} onRemove={onRemove} />);
    expect(screen.getByText('Notice')).toBeInTheDocument();
    expect(screen.getByText('notice text')).toBeInTheDocument();
    expect(screen.getByText('admin')).toBeInTheDocument();
    expect(screen.getByText('1')).toBeInTheDocument();
  });

  it('renders question type data', async () => {
    render(<BoardItem post={question} admin={admin} onRemove={onRemove} />);
    expect(screen.getByText('question text')).toBeInTheDocument();
    expect(screen.getByText('작성자')).toBeInTheDocument();
    expect(screen.getByText('2')).toBeInTheDocument();
  });

  it('renders post type data', async () => {
    render(<BoardItem post={post} admin={admin} onRemove={onRemove} />);
    expect(screen.getByText('Request')).toBeInTheDocument();
    expect(screen.getByText('post text')).toBeInTheDocument();
    expect(screen.getByText('작성자')).toBeInTheDocument();
    expect(screen.getByText('3')).toBeInTheDocument();
    expect(screen.getByText('N')).toBeInTheDocument();
  });

  it('calls onRemove with post when deleteButton clicked', async () => {
    render(<BoardItem post={question} admin={admin} onRemove={onRemove} />);
    window.prompt = jest.fn().mockImplementation(() => '2');
    const deleteButton = screen.getByText('delete');
    await userEvent.click(deleteButton);

    expect(onRemove).toHaveBeenCalledTimes(1);
    expect(onRemove).toHaveBeenCalledWith(question);
  });
});
