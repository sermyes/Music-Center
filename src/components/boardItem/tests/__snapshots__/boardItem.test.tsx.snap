// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Item renders correctly 1`] = `
<li
  className="container"
>
  <div
    className="notice"
  >
    <span
      className="type"
    >
      Notice
    </span>
    <span
      className="noticeContent"
    >
      notice text
    </span>
    <p
      className="adminWrapper"
    >
      <span
        className="name"
      >
        1
      </span>
      <span
        className="admin"
      >
        admin
      </span>
    </p>
    <div
      className="option"
    >
      <button
        className="optionBtn optionBtn"
      >
        <span
          className="elipsis"
        />
        <span
          className="elipsis"
        />
        <span
          className="elipsis"
        />
      </button>
      <button
        className="delBtn deleteBtn"
        onClick={[Function]}
      >
        delete
      </button>
    </div>
  </div>
</li>
`;
